### PR TITLE
 feat(sol): sol updates

### PR DESCRIPTION
**Pull request by Mahendra-intel**

### CODE REVIEW

File: src/app/devices/kvm/kvm.component.spec.ts [hunk 1]
```diff
@@ -2,36 +2,45 @@
...
-    component = fixture.componentInstance
-    fixture.detectChanges()
+    // fixture = TestBed.createComponent(KvmComponent)
+    // component = fixture.componentInstance
+    // fixture.detectChanges()
   })
 
   it('should create', () => {
-    expect(component).toBeTruthy()
+    // expect(component).toBeTruthy()
```
Comment (rsdmike) [inline src/app/devices/kvm/kvm.component.spec.ts:44]: why is this all commented out?

File: src/app/devices/kvm/kvm.component.ts [hunk 2]
```diff
@@ -31,29 +39,88 @@ export class KvmComponent implements OnInit, AfterViewInit {
...
+  constructor (public snackBar: MatSnackBar, public dialog: MatDialog, private readonly devicesService: DevicesService) {
 
   }
 
+  // ngOnChanges ():void{
+  //   console.log("ngOnChanges kvm")
+  // }
+
   ngOnInit (): void {
+    console.log('ngOnInit kvm')
```
Comment (rsdmike) [inline src/app/devices/kvm/kvm.component.ts:55]: remove console statements

File: src/app/devices/kvm/kvm.component.ts [hunk 3]
```diff
@@ -31,29 +39,88 @@ export class KvmComponent implements OnInit, AfterViewInit {
...
-    this.instantiate()
-    // this.setAmtFeatures()
-    this.autoConnect()
+  ngDoCheck (): void {
+    if (this.selectedAction !== this.previousAction) {
+      this.previousAction = this.selectedAction
+    }
   }
 
+  // ngAfterContentInit(){
```
Comment (rsdmike) [inline src/app/devices/kvm/kvm.component.ts:101]: remove commented code

File: src/app/devices/kvm/kvm.component.ts [hunk 4]
```diff
@@ -114,4 +193,9 @@ export class KvmComponent implements OnInit, AfterViewInit {
       this.showKvmChange.emit(false)
     }
   }
+
+  ngOnDestroy = (): void => {
+    console.log('ngOnDestroy kvm')
```
Comment (rsdmike) [inline src/app/devices/kvm/kvm.component.ts:198]: remove console statements

File: src/app/devices/sol/sol.component.ts [hunk 5]
```diff
@@ -21,42 +31,67 @@ export class SolComponent implements OnInit {
...
-      e.preventDefault()
-      if (e.ctrlKey && e.shiftKey && e.keyCode === 67) {
-        return navigator.clipboard.writeText(this.term.getSelection())
-      } else if (e.ctrlKey && e.shiftKey && e.keyCode === 86) {
-        return navigator.clipboard.readText().then(text => {
-          this.handleKeyPress(text)
+    this.setAmtFeatures()
+    this.deviceService.getPowerState(this.uuid).pipe(
+      catchError(err => {
+        console.log(err, 'coming error case')
```
Comment (rsdmike) [inline src/app/devices/sol/sol.component.ts:48]: remove console log

File: src/app/devices/sol/sol.component.ts [hunk 6]
```diff
@@ -21,42 +31,67 @@ export class SolComponent implements OnInit {
...
+        this.snackBar.open($localize`Error enabling sol`, undefined, SnackbarDefaults.defaultError)
```
Comment (rsdmike) [inline src/app/devices/sol/sol.component.ts:49]: Shouldnt this be an error related to getting the power state?

File: src/app/devices/sol/sol.component.ts [hunk 7]
```diff
@@ -74,14 +109,31 @@ export class SolComponent implements OnInit {
...
       fontWeight: 'bold'
     })
+    this.term.open(this.container)
+    this.term.onData((data: any) => {
+      this.handleKeyPress(data)
+    })
+    this.term.attachCustomKeyEventHandler((e: any) => {
+      e.stopPropagation()
+      e.preventDefault()
+      if (e.ctrlKey && e.shiftKey && e.keyCode === 67) {
```
Comment (rsdmike) [inline src/app/devices/sol/sol.component.ts:119]: leverage @angular/cdk/keycodes for constants (https://github.com/angular/components/blob/master/src/cdk/keycodes/keycodes.ts)  should be C

File: src/app/devices/sol/sol.component.ts [hunk 8]
```diff
@@ -74,14 +109,31 @@ export class SolComponent implements OnInit {
...
+        return navigator.clipboard.writeText(this.term.getSelection())
+      } else if (e.ctrlKey && e.shiftKey && e.keyCode === 86) {
```
Comment (rsdmike) [inline src/app/devices/sol/sol.component.ts:121]: leverage @angular/cdk/keycodes for constants (https://github.com/angular/components/blob/master/src/cdk/keycodes/keycodes.ts)  should be V

File: src/environments/environment.ts [hunk 9]
```diff
@@ -8,8 +8,8 @@
 
 export const environment = {
   production: false,
-  mpsServer: 'https://localhost:3000',
-  rpsServer: 'https://localhost:8081',
+  mpsServer: 'https://52.237.119.140:3000',
```
Comment (rsdmike) [inline src/environments/environment.ts:11]: undo changes here

File: src/app/shared/power-up-alert/power-up-alert.component.html [hunk 10]
```diff
@@ -0,0 +1,8 @@
+<h2 mat-dialog-title>Your device is not powered on.</h2>
```
Comment (rsdmike) [inline src/app/shared/power-up-alert/power-up-alert.component.html:1]: remove period from title

File: src/app/devices/sol/sol.component.ts [hunk 11]
```diff
@@ -21,42 +31,67 @@ export class SolComponent implements OnInit {
...
+      finalize(() => {
+      })
+    ).subscribe(data => {
+      this.powerState = data
+      this.isPoweredOn = this.checkPowerStatus()
+      if (!this.isPoweredOn) {
+        const dialog = this.dialog.open(PoweralertComponent)
+        dialog.afterClosed().subscribe(result => {
+          if (result) {
+            this.deviceService.sendPowerAction(this.uuid, 2).pipe().subscribe(data => {
```
Comment (rsdmike) [inline src/app/devices/sol/sol.component.ts:61]: need to add catchError here

File: src/app/devices/sol/poweralert/poweralert.component.ts [hunk 12]
```diff
@@ -0,0 +1,11 @@
+import { Component, OnInit } from '@angular/core'
+
+@Component({
+  selector: 'app-poweralert',
+  templateUrl: './poweralert.component.html',
+  styleUrls: ['./poweralert.component.scss']
+})
+export class PoweralertComponent implements OnInit {
```
Comment (rsdmike) [inline src/app/devices/sol/poweralert/poweralert.component.ts:8]: this should be renamed to  `PowerAlertComponent`

File: src/app/devices/sol/poweralert/poweralert.component.html [hunk 13]
```diff
@@ -0,0 +1,8 @@
+<h2 mat-dialog-title>Device is disconnected</h2>
+<mat-dialog-content class="mat-typography">
+    Are you sure you to connect the device ?
```
Comment (rsdmike) [inline src/app/devices/sol/poweralert/poweralert.component.html:3]: remove space after device

Comment (rsdmike) [reply src/app/devices/sol/poweralert/poweralert.component.html:3]: This doesn't read correctly. Is it supposed to say "Are you sure you want to connect to the device?"

**Review by rsdmike — CHANGES_REQUESTED** [review]

File: src/app/devices/sol/poweralert/poweralert.component.html (hunk 13 above)
Comment (Mahendra-intel) [reply src/app/devices/sol/poweralert/poweralert.component.html:3]: done

File: src/app/devices/sol/poweralert/poweralert.component.ts (hunk 12 above)
Comment (Mahendra-intel) [reply src/app/devices/sol/poweralert/poweralert.component.ts:8]: done

File: src/app/devices/sol/sol.component.ts (hunk 11 above)
Comment (Mahendra-intel) [reply src/app/devices/sol/sol.component.ts:61]: done

File: src/environments/environment.ts (hunk 9 above)
Comment (Mahendra-intel) [reply src/environments/environment.ts:11]: done

File: src/app/devices/kvm/kvm.component.ts (hunk 2 above)
Comment (Mahendra-intel) [reply src/app/devices/kvm/kvm.component.ts:55]: done

File: src/app/devices/kvm/kvm.component.ts (hunk 3 above)
Comment (Mahendra-intel) [reply src/app/devices/kvm/kvm.component.ts:101]: done

File: src/app/devices/kvm/kvm.component.ts (hunk 4 above)
Comment (Mahendra-intel) [reply src/app/devices/kvm/kvm.component.ts:198]: done

File: src/app/devices/sol/sol.component.ts (hunk 5 above)
Comment (Mahendra-intel) [reply src/app/devices/sol/sol.component.ts:48]: done

File: src/app/devices/sol/sol.component.ts (hunk 6 above)
Comment (Mahendra-intel) [reply src/app/devices/sol/sol.component.ts:49]: done

File: src/app/devices/kvm/kvm.component.spec.ts (hunk 1 above)
Comment (Mahendra-intel) [reply src/app/devices/kvm/kvm.component.spec.ts:44]: will be fixed in kvm updates PR

File: src/app/shared/power-up-alert/power-up-alert.component.html (hunk 10 above)
Comment (Mahendra-intel) [reply src/app/shared/power-up-alert/power-up-alert.component.html:1]: done

File: src/app/devices/sol/sol.component.ts (hunk 7 above)
Comment (Mahendra-intel) [reply src/app/devices/sol/sol.component.ts:119]: done

File: src/app/devices/sol/sol.component.ts (hunk 8 above)
Comment (Mahendra-intel) [reply src/app/devices/sol/sol.component.ts:121]: done

**Review by rsdmike — APPROVED** [review]

**Pull request merged**